contracts/AgentMech.sol

```diff
@@ -79,28 +79,47 @@ contract AgentMech is ERC721Mech {
     event RevokeRequest(address indexed sender, uint256 requestId);
     event PriceUpdated(uint256 price);
 
+    enum RequestStatus {
+        DoesNotExist,
+        Requested,
+        Delivered
+    }
+
     // Agent mech version number
     string public constant VERSION = "1.1.0";
+    // Domain separator type hash
+    bytes32 public constant DOMAIN_SEPARATOR_TYPE_HASH =
+    keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)");
+    // Original domain separator value
+    bytes32 public immutable domainSeparator;
+    // Original chain Id
+    uint256 public immutable chainId;
```

Comment:
Reverted back the logic for when agent mechs process requests not via a marketplace

Reply:
Why?

Reply:
request Ids must be unique and not intersect with the Marketplace ones.